yt/config.py

```diff
@@ -67,14 +67,21 @@
 )
 
 
-CONFIG_DIR = os.environ.get(
-    "XDG_CONFIG_HOME", os.path.join(os.path.expanduser("~"), ".config", "yt")
-)
-if not os.path.exists(CONFIG_DIR):
-    try:
-        os.makedirs(CONFIG_DIR)
-    except OSError:
-        warnings.warn("unable to create yt config directory")
+def config_dir():
+    conf_dir = os.environ.get(
+        "XDG_CONFIG_HOME", os.path.join(os.path.expanduser("~"), ".config", "yt")
+    )
```

Comment:
I have to run I can't review fully now but I think there's a mistake here
```suggestion
    XDG = os.environ.get("XDG_CONFIG_HOME",  os.path.join(os.path.expanduser("~"), ".config"))
    conf_dir = os.path.join(os.path.expanduser(XDG, "yt")
```